From ba0cabff401a854d2714e01febbd7b30203a69d8 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Thu, 28 Mar 2013 13:10:31 +0100
Subject: [PATCH 557/715] drm/rcar-du: Add XRGB1555 format support

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit ab636bbf14d93ba96002af8bf06cdace1e1f6531)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit 0e660ceeefe519cbc8905415b521b21d96f4c6fa)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_kms.c   |    6 ++++++
 drivers/gpu/drm/rcar-du/rcar_du_plane.c |    2 ++
 2 files changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_kms.c b/drivers/gpu/drm/rcar-du/rcar_du_kms.c
index 3060ab7..ce8bf2d 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_kms.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_kms.c
@@ -47,6 +47,12 @@ static const struct rcar_du_format_info rcar_du_format_infos[] = {
 		.pnmr = PnMR_SPIM_ALP | PnMR_DDDF_ARGB,
 		.edf = PnDDCR4_EDF_NONE,
 	}, {
+		.fourcc = DRM_FORMAT_XRGB1555,
+		.bpp = 16,
+		.planes = 1,
+		.pnmr = PnMR_SPIM_TP | PnMR_DDDF_ARGB,
+		.edf = PnDDCR4_EDF_NONE,
+	}, {
 		.fourcc = DRM_FORMAT_XRGB8888,
 		.bpp = 32,
 		.planes = 1,
diff --git a/drivers/gpu/drm/rcar-du/rcar_du_plane.c b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
index 9292c7a..6f1dd84 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_plane.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
@@ -168,6 +168,7 @@ static void rcar_du_plane_setup_mode(struct rcar_du_plane *plane,
 		break;
 
 	case DRM_FORMAT_ARGB1555:
+	case DRM_FORMAT_XRGB1555:
 		colorkey = ((plane->colorkey & 0xf80000) >> 9)
 			 | ((plane->colorkey & 0x00f800) >> 6)
 			 | ((plane->colorkey & 0x0000f8) >> 3);
@@ -397,6 +398,7 @@ static const struct drm_plane_funcs rcar_du_plane_funcs = {
 static const uint32_t formats[] = {
 	DRM_FORMAT_RGB565,
 	DRM_FORMAT_ARGB1555,
+	DRM_FORMAT_XRGB1555,
 	DRM_FORMAT_XRGB8888,
 	DRM_FORMAT_ARGB8888,
 	DRM_FORMAT_UYVY,
-- 
1.7.10.4

